pkg/envoy/lds/rbac_test.go

```diff
@@ -0,0 +1,154 @@
+package lds
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	xds_rbac "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+
+	"github.com/openservicemesh/osm/pkg/catalog"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+func TestBuildInboundRBACPolicies(t *testing.T) {
+	assert := assert.New(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockCatalog := catalog.NewMockMeshCataloger(mockCtrl)
+	proxySvcAccount := service.K8sServiceAccount{Name: "sa-1", Namespace: "ns-1"}
+
+	lb := &listenerBuilder{
+		meshCatalog: mockCatalog,
+		svcAccount:  proxySvcAccount,
+	}
+
+	testCases := []struct {
+		name                      string
+		allowedInboundSvcAccounts []service.K8sServiceAccount
+		expectedPrincipals        []string
+		expectErr                 bool
+	}{
+		{
+			"multiple client allowed",
+			[]service.K8sServiceAccount{
+				{Name: "sa-2", Namespace: "ns-2"},
+				{Name: "sa-3", Namespace: "ns-3"},
+			},
+			[]string{
+				"sa-2.ns-2.cluster.local",
+				"sa-3.ns-3.cluster.local",
+			},
+			false, // no error
+		},
+		{
+			"no clients allowed",
+			[]service.K8sServiceAccount{},
+			[]string{},
+			false, // no error
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("Testing test case %d: %s", i, tc.name), func(t *testing.T) {
+			// Mock the calls to catalog
+			mockCatalog.EXPECT().ListAllowedInboundServiceAccounts(lb.svcAccount).Return(tc.allowedInboundSvcAccounts, nil).Times(1)
+
+			// Test the RBAC policies
+			networkRBAC, err := lb.buildInboundRBACPolicies()
+			assert.Equal(err != nil, tc.expectErr)
+
+			assert.Equal(networkRBAC.Rules.GetAction(), xds_rbac.RBAC_ALLOW)
+
+			rbacPolicies := networkRBAC.Rules.Policies
+
+			// Expect 1 policy per client principal
+			assert.Len(rbacPolicies, len(tc.expectedPrincipals))
+
+			// Loop through the policies and ensure there is a policy corresponding to each principal
+			var actualPrincipals []string
+			for _, policy := range rbacPolicies {
+				principalName := policy.Principals[0].GetOrIds().Ids[0].GetAuthenticated().PrincipalName.GetExact()
+				actualPrincipals = append(actualPrincipals, principalName)
+
+				assert.Len(policy.Permissions, 1) // Any permission
+				assert.True(policy.Permissions[0].GetAny())
+			}
+			assert.ElementsMatch(tc.expectedPrincipals, actualPrincipals)
+		})
+	}
+}
+
+func TestBuildRBACFilter(t *testing.T) {
+	assert := assert.New(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockCatalog := catalog.NewMockMeshCataloger(mockCtrl)
+	proxySvcAccount := service.K8sServiceAccount{Name: "sa-1", Namespace: "ns-1"}
+
+	lb := &listenerBuilder{
```

Comment:
Why not use `newListenerBuilder()`?

Reply:
Because this allows to fill in minimal stuff into the object needed to build the filter. When a new field is added to the struct, if the test fails it helps find dependency issues faster.